pkg/cmd/pipeline/start.go

```diff
@@ -0,0 +1,177 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	errNoPipeline      = errors.New("missing pipeline name")
+	errInvalidPipeline = errors.New("invalid pipeline name")
+)
+
+const (
+	invalidResource = "invalid resource parameter: "
+	invalidParam    = "invalid param parameter: "
+)
+
+type startOptions struct {
+	Params             []string
+	Resources          []string
+	ServiceAccountName string
+}
+
+// NameArg validates that the first argument is a valid pipeline name
+func NameArg(p cli.Params) cobra.PositionalArgs {
+
+	return func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return errNoPipeline
+		}
+
+		c, err := p.Clients()
+		if err != nil {
+			return err
+		}
+
+		name, ns := args[0], p.Namespace()
+		_, err = c.Tekton.TektonV1alpha1().Pipelines(ns).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return errInvalidPipeline
+		}
+
+		return nil
+	}
+
+}
+
+func startCommand(p cli.Params) *cobra.Command {
+	var (
+		res    []string
+		params []string
+		svc    string
+	)
+
+	c := &cobra.Command{
+		Use:     "start",
+		Aliases: []string{"trigger"},
+		Short:   "Start pipelines",
+		Example: `
+  # start pipeline foo by creating a pipelienrun named "foo-run-xyz123" from the namespace "bar"
+    tkn pipeline start foo --param NAME=VALUE --resource source=scaffold-git  -s ServiceAccountName  -n bar
+   `,
+		SilenceUsage: true,
+		Args:         NameArg(p),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			opt := startOptions{
+				Resources:          res,
+				Params:             params,
+				ServiceAccountName: svc,
+			}
+			return startPipeline(cmd.OutOrStdout(), opt, p, args[0])
+		},
+	}
+
+	c.Flags().StringSliceVarP(&res, "resource", "r", []string{}, "pass the resource name and ref")
+	c.Flags().StringSliceVarP(&params, "param", "p", []string{}, "pass the param")
+	c.Flags().StringVarP(&svc, "serviceaccount", "s", svc, "pass the serviceaccount name")
```

Comment:
I'm sorry, didn't realize now PipelineRun has an ability to provide multiple SA with this PR: 
 https://github.com/tektoncd/pipeline/pull/902

Reply:
anyways either we can fix it in this PR or in follow up issue

Reply:
That's a bit different from passing svc. It will require a map of svc and tasks. https://github.com/tektoncd/pipeline/blob/f452c95fe0718f4c7fb1d9250f75d95dfa95c971/pkg/apis/pipeline/v1alpha1/pipelinerun_types.go#L52

Reply:
Merge this. WIll send a follow up pr.

Reply:
We need to update vendor version also.